src/Microsoft.DotNet.Interactive.HttpRequest.Tests/HttpRequestKernelTests.cs

```diff
@@ -0,0 +1,256 @@
+    {
+        using var kernel = new HttpRequestKernel();
+
+        var result = await kernel.SendAsync(new SubmitCode("get  /relativePath"));
+
+        var events = result.KernelEvents.ToSubscribedList();
+
+        var error =  events.Should().ContainSingle<CommandFailed>().Which;
+
+        error.Message.Should().Contain("Cannot use relative path /relativePath without a base address.");
+    }
+
+    [Fact]
+    public async Task ignores_base_address_when_using_absolute_paths()
+    {
+        HttpRequestMessage request = null;
+        var handler = new InterceptingHttpMessageHandler((message, _) =>
+        {
+            request = message;
+            var response = new HttpResponseMessage(HttpStatusCode.OK);
+            return Task.FromResult(response);
+        });
+        var client = new HttpClient(handler);
+        using var kernel = new HttpRequestKernel(client: client);
+        kernel.BaseAddress = new Uri("http://baseAddress.com");
+        
+        var result = await kernel.SendAsync(new SubmitCode("get  https://anotherlocation.com/endpoint"));
+
+        var events = result.KernelEvents.ToSubscribedList();
+
+        events.Should().NotContainErrors();
+
+        request.RequestUri.Should().Be("https://anotherlocation.com/endpoint");
+    }
+
+    [Fact]
+    public async Task can_replace_symbols()
+    {
+        HttpRequestMessage request = null;
+        var handler = new InterceptingHttpMessageHandler((message, _) =>
+        {
+            request = message;
+            var response = new HttpResponseMessage(HttpStatusCode.OK);
+            return Task.FromResult(response);
+        });
+        var client = new HttpClient(handler);
+        using var kernel = new HttpRequestKernel(client: client);
+     
+        kernel.SetValue("my_host", "my.host.com");
+
+        var result = await kernel.SendAsync(new SubmitCode("get  https://{{my_host}}:1200/endpoint"));
+
+        var events = result.KernelEvents.ToSubscribedList();
+
+        events.Should().NotContainErrors();
+
+        request.RequestUri.Should().Be("https://my.host.com:1200/endpoint");
+    }
+
+    [Fact]
+    public async Task can_use_base_address_to_resolve_host_symbol()
+    {
+        HttpRequestMessage request = null;
+        var handler = new InterceptingHttpMessageHandler((message, _) =>
+        {
+            request = message;
+            var response = new HttpResponseMessage(HttpStatusCode.OK);
+            return Task.FromResult(response);
+        });
+        var client = new HttpClient(handler);
+        using var kernel = new HttpRequestKernel(client: client);
+        kernel.BaseAddress = new Uri("http://baseAddress.com");
+
+        var result = await kernel.SendAsync(new SubmitCode("get  https://{{host}}:1200/endpoint"));
+
+        var events = result.KernelEvents.ToSubscribedList();
+
+        events.Should().NotContainErrors();
+
+        request.RequestUri.Should().Be("https://baseaddress.com:1200/endpoint");
```

Comment:
I'm not understanding this one.  Shouldn't the result be `https://http://baseAddress.com:1200/endpoint`?